src/main/kotlin/uk/gov/justice/digital/hmpps/sendlegalmailtoprisonsapi/prisonersearch/PrisonerSearchService.kt

```diff
@@ -0,0 +1,34 @@
+package uk.gov.justice.digital.hmpps.sendlegalmailtoprisonsapi.prisonersearch
+
+import mu.KotlinLogging
+import org.springframework.http.HttpStatus
+import org.springframework.stereotype.Service
+import org.springframework.web.reactive.function.client.WebClientResponseException
+import uk.gov.justice.digital.hmpps.sendlegalmailtoprisonsapi.barcode.BarcodeRecipient
+import java.util.stream.Collectors
+
+private val log = KotlinLogging.logger {}
+
+@Service
+class PrisonerSearchService(private val prisonerSearchClient: PrisonerSearchClient) {
+  fun lookupPrisoner(barcodeRecipient: BarcodeRecipient) {
+    val prisonerSearchRequest = prisonerSearchRequest(barcodeRecipient)
+    try {
```

Comment:
All I want to do at this point is call `match-prisoners` and log the number of potential matches, or something that says the user token doesnt have the right roles. Once we start seeing some log data we can move on from there ....